internal/providers/vault/README.md

```diff
@@ -0,0 +1,88 @@
+# Vault Provider
+
+The Vault provider for Secretless can fetch secrets from configured backends 
+in [HashiCorp Vault](https://www.vaultproject.io). The provider is based on 
+the [Vault API client](https://pkg.go.dev/github.com/hashicorp/vault/api) in Go. 
+It reads values from the configured paths.
+
+## Usage Documentation
+
+The Vault provider is configured in the `secretless.yml`. Using
+`from: vault` and `get: /path/to/secret/in/vault` for the credentials in the 
```

Comment:
You should use a code block for the config here probably

Reply:
The Vault provider is configured in the `secretless.yml`. Using
    ```yaml
    from: vault
    get: /path/to/secret/in/vault
    ```
    for the credentials in the ...